src/components/Reports/GoalCalculator/Shared/GoalCalculatorContext.tsx

```diff
@@ -112,6 +122,55 @@ export const GoalCalculatorProvider: React.FC<Props> = ({ children }) => {
     setIsDrawerOpen((prev) => !prev);
   }, []);
 
+  const getFamilySections = useCallback(
+    (budgetFamily: BudgetFamilyFragment): SectionItem[] => {
+      return (
+        budgetFamily.primaryBudgetCategories.map((category) => ({
+          title: category.label,
+          complete:
+            category.directInput !== null ||
+            category.subBudgetCategories.some(
+              (category) => category.amount > 0,
+            ),
+        })) ?? []
+      );
+    },
+    [],
+  );
+
+  const percentComplete = useMemo(() => {
+    const { data } = goalCalculationResult;
+    if (!data?.goalCalculation) {
+      return 0;
+    }
+
+    const allFamilies = [
+      data.goalCalculation.householdFamily,
+      data.goalCalculation.ministryFamily,
+    ].filter(Boolean);
```

Comment:
Families aren't nullable.

```suggestion
    ];
```